src/algorithms/graphs/depthFirstSearch.java

```diff
@@ -0,0 +1,84 @@
+package src.algorithms.graphs;
+
+import java.util.*;
+import src.algorithms.graphs.util.TreeNode;
+
+/**
+ * Implementation of DFS
+ *
+ * Depth-First search is a graph traversal algorithm that utilizes a stack (FIFO) data structure
+ * It is useful in finding a shortest-path solution (IN TREES), or a single solution in graphs
+ * This method is also used to obtain order-traversals of trees.
+ *
+ * In general, DFS works as such:
```

Comment:
This is a really good explanation! Thanks.

The only area for improvement is perhaps the use of preOrder as the example. preOrder is a special type of DFS search that doesn't require the seen HashSet variable to be maintained since each node will also be seen once. Further, a tree is also a special kind of graph. Afraid this might cause confusion, or throw off new coders because its deviating from what was discussed.

I didn't think of this before but looking at your code helps me to think of a better structure. We can do BFS and DFS on general connected graphs (a small 1-liner to explain for unconnected graphs, simply run the algorithm agn for unvisited nodes). Then, we can also write the algorithm like what you did here for Pre-order, Post-order, In-order which are special forms of DFS and Level-order which is a special form of BFS